addons/sale_expense/tests/test_sale_expense.py

```diff
@@ -97,3 +97,57 @@ def test_sale_expense(self):
         # both expenses should be invoiced
         inv = so._create_invoices()
         self.assertEqual(inv.amount_untaxed, 621.5 + (prod_exp_2.list_price * 100.0), 'Sale Expense: invoicing of expense is wrong')
+
+    def test_expense_multi_id_analytic_distribution(self):
+        """
+        Test conversion of analytic_distribution dict into account numbers when a hr.expense.sheet containing
+        a hr.expense with an analytic_distribution having 2+ account ids
+        """
+        product = self.env['product.product'].create({
+            'name': 'test product',
+            'can_be_expensed': True,
+            'detailed_type': 'service',
+            'invoice_policy': 'order',
+            'standard_price': 100,
+            'expense_policy': 'cost',
+        })
+
+        sale_order = self.env['sale.order'].create({
+            'partner_id': self.partner_a.id,
+            'partner_invoice_id': self.partner_a.id,
+            'partner_shipping_id': self.partner_a.id,
+            'order_line': [Command.create({
+                'product_id': self.product_b.id,
+            })],
+        })
+        sale_order.action_confirm()
+        sale_order._create_invoices()
+
+        analytic_account_3 = self.env['account.analytic.account'].create({
+            'name': 'analytic_account_3',
+            'plan_id': self.analytic_plan.id,
+        })
+
+        expense_sheet = self.env['hr.expense.sheet'].create({
+            'name': 'Test Expense Report',
+            'employee_id': self.expense_employee.id,
+            'company_id': self.company_data['company'].id,
+            'expense_line_ids': [Command.create({
+                'employee_id': self.expense_employee.id,
+                'product_id': product.id,
+                'quantity': 1000.00,
+                'date': fields.Date.today(),
+                'company_id': self.company_data['company'].id,
+                'currency_id': self.company_data['currency'].id,
+                'analytic_distribution': {
+                    f'{self.analytic_account_1.id},{self.analytic_account_2.id}': 100,
+                    f'{analytic_account_3.id}': 40,
```

Comment:
```suggestion
                    f'{self.analytic_account_1.id},{self.analytic_account_2.id}': 60,
                    f'{analytic_account_3.id}': 40,
```
You're not supposed to have more than 100% in your distribution :wink: